### PR TITLE
Cldc 1790 sales validation content updates

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
image

### CODE REVIEW

File: app/helpers/interruption_screen_helper.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ def display_informative_text(informative_text, lettings_log)
       value = if argument["label"]
                 pre_casing_value = lettings_log.form.get_question(argument["key"], lettings_log).answer_label(lettings_log)
                 pre_casing_value.downcase
+              elsif argument["currency"]
+                ["£", ActionController::Base.helpers.number_to_currency(lettings_log.public_send(argument["key"]), delimiter: ",", format: "%n")].join("")
```
Comment (arfacamble) [inline app/helpers/interruption_screen_helper.rb:11]: we should discuss this so I understand properly what it's doing.

also default argument of join is "" so we can remove that

Comment (arfacamble) [reply app/helpers/interruption_screen_helper.rb:11]: also while discussing this we should discuss "label"

File: app/models/form/sales/pages/buyer1_income_value_check.rb [hunk 2]
```diff
@@ -6,6 +6,22 @@ def initialize(id, hsh, subsection)
...
+          "i18n_template" => "income1",
+        },
+        {
+          "key" => "allowed_income_soft_min",
+          "label" => false,
+          "i18n_template" => "soft_min",
+          "currency" => true,
+        },
+      ],
+    }
```
Comment (arfacamble) [inline app/models/form/sales/pages/buyer1_income_value_check.rb:24]: We should remove and I should reimplement on my branch in my similar way.

File: app/models/form/sales/questions/buyer1_income_value_check.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ def initialize(id, hsh, page)
     super
     @id = "income1_value_check"
     @check_answer_label = "Income confirmation"
-    @header = "Are you sure this income is correct?"
+    @header = "Are you sure this is correct?"
```
Comment (arfacamble) [inline app/models/form/sales/questions/buyer1_income_value_check.rb:6]: I need to do this on my branch

File: app/models/sales_log.rb [hunk 4]
```diff
@@ -232,4 +232,8 @@ def buyers_age_for_old_persons_shared_ownership_invalid?
   def ages_unknown_or_under_64?(person_indexes)
     person_indexes.all? { |person_num| self["age#{person_num}"].present? && self["age#{person_num}"] < 64 || self["age#{person_num}_known"] == 1 }
   end
+
+  def allowed_income_soft_min
+    ALLOWED_INCOME_RANGES[ecstat1].soft_min
+  end
```
Comment (arfacamble) [inline app/models/sales_log.rb:238]: remove I have implemented similar but more flexible

File: app/models/validations/sales/financial_validations.rb [hunk 5]
```diff
@@ -4,18 +4,18 @@ module Validations::Sales::FinancialValidations
...
+        record.errors.add :ecstat1, I18n.t("validations.financial.income1.over_hard_max.inside_london")
+        record.errors.add :ownershipsch, I18n.t("validations.financial.income1.over_hard_max.inside_london")
+        record.errors.add :la, I18n.t("validations.financial.income1.over_hard_max.inside_london")
+        record.errors.add :postcode_full, I18n.t("validations.financial.income1.over_hard_max.inside_london")
+      elsif !record.london_property? && record.income1 > 80_000
+        record.errors.add :income1, I18n.t("validations.financial.income1.over_hard_max.outside_london")
+        record.errors.add :ecstat1, I18n.t("validations.financial.income1.over_hard_max.outside_london")
+        record.errors.add :ownershipsch, I18n.t("validations.financial.income1.over_hard_max.outside_london")
+        record.errors.add :la, I18n.t("validations.financial.income1.over_hard_max.outside_london")
+        record.errors.add :postcode_full, I18n.t("validations.financial.income1.over_hard_max.outside_london")
```
Comment (arfacamble) [inline app/models/validations/sales/financial_validations.rb:18]: remove all changes, I have completely changed this file so just avoid conflicts

File: app/models/validations/sales/soft_validations.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@ module Validations::Sales::SoftValidations
   def income1_under_soft_min?
     return false unless ecstat1 && income1 && ALLOWED_INCOME_RANGES[ecstat1]
 
-    income1 < ALLOWED_INCOME_RANGES[ecstat1][:soft_min]
+    income1 < ALLOWED_INCOME_RANGES[ecstat1] [:soft_min]
```
Comment (arfacamble) [inline app/models/validations/sales/soft_validations.rb:13]: I feel like this change shouldn't be here anyway. Shouldn't cause conflicts, but is there a reason this is here?

File: config/locales/en.yml [hunk 7]
```diff
@@ -215,7 +215,9 @@ en:
         freq_missing: "Select how often the household receives income"
         earnings_missing: "Enter how much income the household has in total"
       income1:
-        over_hard_max: "Income must be lower than £%{hard_max}"
+        over_hard_max:
+          outside_london: "Income must not exceed £80,000 for properties outside London local authorities"
+          inside_london: "Income must not exceed £90,000 for properties within London local authorities"
```
Comment (arfacamble) [inline config/locales/en.yml:220]: remove, will cause conflicts, I have reimplemented on my branch

File: config/locales/en.yml [hunk 8]
```diff
@@ -389,7 +391,7 @@ en:
 
     declaration:
       missing: "You must show the DLUHC privacy notice to the tenant before you can submit this log."
-    
+
```
Comment (arfacamble) [inline config/locales/en.yml:394]: is this some project wide yml formatting rule that I have been ignoring?

File: config/locales/en.yml [hunk 9]
```diff
@@ -446,6 +448,7 @@ en:
         message: "Net income is lower than expected based on the lead tenant’s working situation. Are you sure this is correct?"
       in_soft_max_range:
         message: "Net income is higher than expected based on the lead tenant’s working situation. Are you sure this is correct?"
+      under_soft_min: "You said income was %{income1}, which is below this working situation's minimum (%{soft_min})"
```
Comment (arfacamble) [inline config/locales/en.yml:451]: remove, duplicate.
Also this related to gross income I believe

File: spec/models/validations/sales/financial_validations_spec.rb [hunk 10]
```diff
@@ -52,15 +52,15 @@
...
+                .to include(match I18n.t("validations.financial.income1.over_hard_max.inside_london"))
             expect(record.errors["ownershipsch"])
-                .to include(match I18n.t("validations.financial.income1.over_hard_max", hard_max: 90_000))
+                .to include(match I18n.t("validations.financial.income1.over_hard_max.inside_london"))
             expect(record.errors["la"])
-                .to include(match I18n.t("validations.financial.income1.over_hard_max", hard_max: 90_000))
+                .to include(match I18n.t("validations.financial.income1.over_hard_max.inside_london"))
             expect(record.errors["postcode_full"])
-                .to include(match I18n.t("validations.financial.income1.over_hard_max", hard_max: 90_000))
+                .to include(match I18n.t("validations.financial.income1.over_hard_max.inside_london"))
```
Comment (arfacamble) [inline spec/models/validations/sales/financial_validations_spec.rb:63]: rollback all changes to this file please

File: config/locales/en.yml (hunk 8 above)
Comment (natdeanlewissoftwire) [reply config/locales/en.yml:394]: maybe - I think this just came from rubocop -A or from RubyMine autocorrection, these lintings haven't been done manually

**Pull request closed without merging**